managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2159,7 +2162,7 @@ void notifyCursors() {
                 break;
             }
 
-            executor.execute(safeRun(waitingCursor::notifyEntriesAvailable));
+            pinnedExecutor.execute(safeRun(waitingCursor::notifyEntriesAvailable));
```

Comment:
Is this required to be on the same executor? 

We're notify multiple cursors that entries are available, this should be able to progress in parallel.

Reply:
There are 2 places will call the `notifyCursors()` method, one is OpAddEntry.safeRun(), it already run the `pinnedExecutor` so don't need to jump again.

Another one is the ledger closed, looks only need to change here.